lib/index.js

```diff
@@ -419,10 +407,77 @@ class Config {
       nopt.clean(obj.data, this.types, this.typeDefs)
 
       nopt.invalidHandler = null
+
+      if (where === 'user') {
```

Comment:
What happens if a user has put a bad config in the global npmrc?

Reply:
hmmmmmmm that's an excellent point. we do have some code that seems to be attempting to keep auth in the `user` config file, but you can absolutely add them to any of the other configs. the project level one having an `${NPM_TOKEN}` reference in it for auth is probably pretty common, even.

i'll expand both this and `repair()` so that they function in the `global`, and `project` configs too. i don't feel like throwing for the `cli` or `env` levels is necessary, and using `builtin` for anything auth related would be an extremely strange decision for someone to make

Reply:
Throwing for `cli` and `env` makes sense to me. The fix isn't as "neat" cause we can't automatically do it for them but can we suggest the fix?